registry/finders.go

```diff
@@ -834,6 +846,23 @@ func GetAppsList(c *Space, opts *AppsListOptions) (int, []*App, error) {
 		}(app)
 	}
 
+	// Alter the cache if there are modifications
+	var updateVersions, updateLatest bool
+	go func() {
```

Comment:
This goroutine runs concurrently to the other goroutine, not after them. So, it doesn't help for avoiding concurrent accesses to the maps.